tools/Google.Cloud.ClientTesting/CodeHealthTester.cs

```diff
@@ -67,5 +67,33 @@ public static void AssertClassesAreSealedOrAbstract(Type sampleType, IEnumerable
             // Force output to show the bad types
             Assert.Equal(new string[0], badTypes.ToList());
         }
+
+        /// <summary>
+        /// Asserts that the assembly containing the given type does not define the disallowed namespaces.
+        /// </summary>
+        public static void AssertNoDisallowedNamespaces(Type sampleType, IEnumerable<string> disallowedNamespeces)
+        {
+            var badNamespaces = sampleType.Assembly.DefinedTypes
+                .Select(t => t.Namespace)
+                .Distinct()
+                .Intersect(disallowedNamespeces);
+
+            // Force output to show the bad namespaces.
+            Assert.Equal(new string[0], badNamespaces.ToList());
```

Comment:
I think `Assert.Empty(badNamespaces)` would show everything, wouldn't it

I think I'd be tempted to list the types themselves - so use `sampleType.Assembly.DefinedTypes.Where(t => disallowedNamespaces.Contains(t.Namespace))` and the equivalent for AssertOnlyAllowed.

Reply:
Done.